# 2. Создайте собственный класс-исключение, обрабатывающий ситуацию деления на нуль.
# Проверьте его работу на данных, вводимых пользователем. При вводе пользователем нуля
# в качестве делителя программа должна корректно обработать эту ситуацию и не завершиться с ошибкой.

class OwnError(Exception):
    def __init__(self, txt):
        self.txt = txt


inp_data = input("Введите деление используя оператор /: ")

try:
    x, y = map(int, inp_data.split('/'))
    if y == 0:
        raise OwnError("Ошибка деления на 0")
except ValueError:
    print("Вы что-то не то ввели")
except OwnError as err:
    print(err)
else:
    print(f"Результат деления: {x / y}")
